payjoin/src/uri.rs

```diff
@@ -301,6 +301,42 @@ enum InternalPjParseError {
     UnsecureEndpoint,
 }
 
+impl<'a> bip21::SerializeParams for &'a Payjoin {
+    type Key = &'static str;
+    type Value = String;
+    type Iterator = std::vec::IntoIter<(Self::Key, Self::Value)>;
+
+    fn serialize_params(self) -> Self::Iterator {
+        match self {
+            Payjoin::Supported(params) => params.serialize_params(),
+            Payjoin::V2Only(params) => params.serialize_params(),
+            Payjoin::Unsupported => vec![].into_iter(),
+        }
+    }
+}
+
+impl<'a> bip21::SerializeParams for &'a PayjoinParams {
+    type Key = &'static str;
+    type Value = String;
+    type Iterator = std::vec::IntoIter<(Self::Key, Self::Value)>;
+
+    fn serialize_params(self) -> Self::Iterator {
+        vec![
+            ("pj", self.endpoint.as_str().to_string()),
+            ("pjos", if self.disable_output_substitution { "1" } else { "0" }.to_string()),
+            #[cfg(feature = "v2")]
+            (
+                "ohttp",
+                bitcoin::base64::encode_config(
+                    self.ohttp_config.as_ref().unwrap().encode().unwrap(),
```

Comment:
not very happy with the second `unwrap()` here, need to look into the internal `encode()` function a bit to understand how it can fail

Reply:
Within library code we should either panic! or .expect to document the good reason why we should be panicking. I agree with your sentiment plus we should not have the first unwrap either. We should not panic here. A valid ohttp_config should always have a valid encoding.